net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/MetadataFactory.java

```diff
@@ -100,6 +134,36 @@ private MetadataFactory() {
     refresh();
   }
 
+  /*
+   * Fetch third party checks extension metadata from YML files.
+   */
+  private static void getThirdPartyModuleExtensionMetadata() {
+    Set<String> eclipseMetaDataFiles = new HashSet<>();
+
+    for (String packageName : sPackageNameSet) {
+      if (packageName.matches(".*\\.checkstyle.*")) {
+        String topMostPackageName = packageName.split("\\.")[0];
+        Reflections reflections = new Reflections(topMostPackageName, new ResourcesScanner());
+        eclipseMetaDataFiles.addAll(reflections
+                             .getResources(Pattern.compile(".*eclipse-metadata.*\\.yml")));
```

Comment:
`Pattern.compile` should be part of class field to be done ones.

Reply:
Done.